src/script/ir_builder/ir/ir.cc

```diff
@@ -69,6 +69,49 @@ void DefFunction(const String& func_name, const BaseFunc& func) {
   }
 }
 
+GlobalVar AddFunction(const BaseFunc& func, String func_name_hint) {
```

Comment:
What's the difference between these two functions and `DeclFunction`, `DefFunction`?

Reply:
The main difference is [FindModuleFrame](https://github.com/apache/tvm/blob/unity/src/script/ir_builder/ir/utils.h#L31-L35) used in `DeclFunction` and `DefFunction`, the two guys are required to be called immediately under I.ir_module(). AddFunction and UpdateFunction don't have this restriction.

Tried to use decl_function in emit_te, ran into error `ValueError: 'I.DeclFunction' must be called immediately under I.ir_module()`

Reply:
Could you enhance those two instead of writing two new ones? 

Excaully `ValueError: 'I.DeclFunction' must be called immediately under I.ir_module()` is not necessary in this case

Reply:
cool, thanks for letting me know that!  Have removed add/update_function. pls take another look